WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsDataSource.swift

```diff
@@ -63,35 +63,41 @@ class ReaderInterestsDataSource {
     }
 
     /// Fetches the interests from the topic service
-    public func reload() {
+    public func reload(completion: (() -> Void)? = nil) {
         interestsService.fetchInterests(success: { [weak self] interests in
```

Comment:
I thought all our API calls already call their success and failure blocks on the main thread, so I wonder if adding `@MainActor` changes anything.

Reply:
Good point. You're right, the API already dispatches the completion blocks to the main thread. My assumption is that the `interests` variable got updated while the collection view is calling `cellForItemAtIndexPath` for the indexes before it got updated, which led to the guard condition. 

However, even if I'm dispatching it again to the main thread—and, even _if_ it works—it doesn't guarantee that the indexes will be correct when `cellForItemAtIndexPath` is called, since we don't know if the collection view still holds onto the "old" `numberOfItemsInSection` value.

Perhaps the best way is to move the implementation to a `UICollectionViewDiffableDataSource`... For now, I'll revert the dispatches to main queue and will just keep the empty cell registration.

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/23158/commits/f560543acaad12b3ae9941832d79f27667509d31.